src/test/java/org/jenkinsci/plugins/workflow/job/WorkflowRunRestartTest.java

```diff
@@ -93,8 +97,45 @@
             assertTrue(p.isResumeBlocked());
             WorkflowRun b = p.getBuildByNumber(1);
             r.waitForCompletion(b);
+            assertFalse(b.executionLoaded);
+            assertTrue(b.completed.get());
             assertFalse(b.isBuilding());
             assertEquals(Result.ABORTED, b.getResult());
+            FlowExecution fe = b.getExecution();
+            assertTrue(b.executionLoaded);
+            assertNotNull(fe.getOwner());
+            assertNull(b.logsToCopy);
+        });
+    }
+
+    @Issue("JENKINS-45585")  // Verifies execution lazy-load
+    @Test public void lazyLoadExecution() {
+        story.thenWithHardShutdown(r -> {
+            WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+            p.setDefinition(new CpsFlowDefinition("echo 'dosomething'", true));
```

Comment:
Do you need to set the durability mode too? Best to not rely on defaults if it is significant here.